src/CoreApi/BlockApi.cs

```diff
@@ -1,113 +1,111 @@
+            CancellationToken cancel = default)
+        {
+            var options = new List<string>();
             if (multiHash != MultiHash.DefaultAlgorithmName ||
                 contentType != Cid.DefaultContentType ||
                 encoding != MultiBase.DefaultAlgorithmName)
             {
                 options.Add($"mhtype={multiHash}");
                 options.Add($"format={contentType}");
                 options.Add($"cid-base={encoding}");
-            }
-            var json = await ipfs.UploadAsync("block/put", cancel, data, null, options.ToArray());
+            }
+            var json = await ipfs.UploadAsync("block/put", cancel, data, name: null, options.ToArray());
             var info = JObject.Parse(json);
-            Cid cid = (string)info["Key"];
-
+            Cid cid = (string)info["Key"]!;
+
             if (pin)
             {
                 await ipfs.Pin.AddAsync(cid, recursive: false, cancel: cancel);
-            }
-
-            return cid;
-        }
-
-        public async Task<IDataBlock> StatAsync(Cid id, CancellationToken cancel = default(CancellationToken))
-        {
-            var json = await ipfs.DoCommandAsync("block/stat", cancel, id);
+            }
+
+            return cid;
+        }
+
+        public async Task<IDataBlock> StatAsync(Cid id, CancellationToken cancel = default)
+        {
+            var json = await ipfs.DoCommandAsync("block/stat", cancel, id);
             var info = JObject.Parse(json);
-            return new Block
-            {
-                Size = (long)info["Size"],
-                Id = (string)info["Key"]
-            };
-        }
-
-        public async Task<Cid> RemoveAsync(Cid id, bool ignoreNonexistent = false, CancellationToken cancel = default(CancellationToken))
-        {
-            var json = await ipfs.DoCommandAsync("block/rm", cancel, id, "force=" + ignoreNonexistent.ToString().ToLowerInvariant());
-            if (json.Length == 0)
-                return null;
-            var result = JObject.Parse(json);
-            var error = (string)result["Error"];
-            if (error != null)
-                throw new HttpRequestException(error);
-            return (Cid)(string)result["Hash"];
+            return new Block
+            {
+                Size = (long?)info["Size"] ?? 0,
+                Id = (string)info["Key"]!
+            };
         }
 
-    }
-
-}
+        public async Task<Cid?> RemoveAsync(Cid id, bool ignoreNonexistent = false, CancellationToken cancel = default)
+        {
+            var json = await ipfs.DoCommandAsync("block/rm", cancel, id, "force=" + ignoreNonexistent.ToString().ToLowerInvariant());
+            if (json.Length == 0)
+                return null;
+            var result = JObject.Parse(json);
+            var error = (string?)result["Error"];
+            if (error is not null)
+                throw new HttpRequestException(error);
+            return (Cid)(string)result["Hash"]!;

```

Comment:
No need to suppress this. We can safely return a null here.
```suggestion
            return (Cid?)(string)result["Hash"];

```